platform/darwin/test/MGLDistanceFormatterTests.m

```diff
@@ -0,0 +1,100 @@
+    XCTAssertEqualObjects([formatter stringFromDistance:0.6096], [self testStringWithDistance:2 formatter:formatter unit:@"feet"]);
+    XCTAssertEqualObjects([formatter stringFromDistance:304.495], [self testStringWithDistance:999 formatter:formatter unit:@"feet"]);
+    XCTAssertEqualObjects([formatter stringFromDistance:1609.34], [self testStringWithDistance:1 formatter:formatter unit:@"mile"]);
+    XCTAssertEqualObjects([formatter stringFromDistance:1609.34], [self testStringWithDistance:1 formatter:formatter unit:@"mile"]);
+    XCTAssertEqualObjects([formatter stringFromDistance:2414.01], [self testStringWithDistance:1.5 formatter:formatter unit:@"miles"]);
+    XCTAssertEqualObjects([formatter stringFromDistance:3218.68], [self testStringWithDistance:2 formatter:formatter unit:@"miles"]);
+}
+
+- (void)testImperialWithYardsAbbreviated {
+    MGLDistanceFormatter *formatter = [[MGLDistanceFormatter alloc] init];
+    formatter.units = MGLDistanceFormatterUnitsImperialWithYards;
+    formatter.unitStyle = MGLDistanceFormatterUnitStyleAbbreviated;
+    
+    XCTAssertEqualObjects([formatter stringFromDistance:0], [self testStringWithDistance:0 formatter:formatter unit:@"yd"]);
+    XCTAssertEqualObjects([formatter stringFromDistance:457.2], [self testStringWithDistance:500 formatter:formatter unit:@"yd"]);
+    XCTAssertEqualObjects([formatter stringFromDistance:913.486], [self testStringWithDistance:999 formatter:formatter unit:@"yd"]);
+    XCTAssertEqualObjects([formatter stringFromDistance:1609.34], [self testStringWithDistance:1 formatter:formatter unit:@"mi"]);
+}
+
+- (void)testImperialWithYards {
+    MGLDistanceFormatter *formatter = [[MGLDistanceFormatter alloc] init];
+    formatter.units = MGLDistanceFormatterUnitsImperialWithYards;
+    formatter.unitStyle = MGLDistanceFormatterUnitStyleFull;
+    
+    XCTAssertEqualObjects([formatter stringFromDistance:0], [self testStringWithDistance:0 formatter:formatter unit:@"yards"]);
+    XCTAssertEqualObjects([formatter stringFromDistance:0.9144], [self testStringWithDistance:1 formatter:formatter unit:@"yard"]);
+    XCTAssertEqualObjects([formatter stringFromDistance:457.2], [self testStringWithDistance:500 formatter:formatter unit:@"yards"]);
+    XCTAssertEqualObjects([formatter stringFromDistance:913.486], [self testStringWithDistance:999 formatter:formatter unit:@"yards"]);
+    XCTAssertEqualObjects([formatter stringFromDistance:1609.34], [self testStringWithDistance:1 formatter:formatter unit:@"mile"]);
+    XCTAssertEqualObjects([formatter stringFromDistance:2414.02], [self testStringWithDistance:1.5 formatter:formatter unit:@"miles"]);
+    XCTAssertEqualObjects([formatter stringFromDistance:3218.68], [self testStringWithDistance:2 formatter:formatter unit:@"miles"]);
+}
+
+- (NSString *)testStringWithDistance:(CLLocationDistance)distance formatter:(MGLDistanceFormatter *)formatter unit:(NSString *)unit {
```

Comment:
It only recognizes methods prefixed with `test` if the return type is void but I agree.